### DISCUSSION TITLE
Vertical vs horizontal tabs

### DISCUSSION THREAD

**Discussion by rafnov** [zen-browser/desktop#2177] [Polls]
What community would like the best? Vertical or horizontal tabs? Or maybe the freedom of choice depending on the need? Should it be configurable or imposed from above?

I totally disagree with the statement from FAQ https://docs.zen-browser.app/faq, 1st question answer: How can I use horizontal tabs?
You DO NOT maximize screen space with vertical tabs, you waste it even more compared to another Firefox clone - Waterfox. See the empty space below the same tabs opened in Zen and the horizontal, humble bar on Waterfox, compare page content. Screenshots attached. Both full screen on 1920x1200 screen (but it does not matter so as they were on 1920x1080, not much of a difference - just 120 pixels), just tried to open same tabs and same amount of them. Both browsers have only 2 extensions installed and enabled: uBlock Origin and Dark Reader (I use only them).
Where is more space wasted?
Of course it is my scenario - few tabs opened, but that's my need. I don't use workspaces, I don't use containters, I don't like to open dosens of tabs. I got needed pages in bookmarks, history or in homepage. I don't like minimizing / hiding tabs or animations to show them. I just need few tabs open at the same time, with their names shown, but not waste 2/3 of space as empty on a tab list, like Zen does.
Therefore I think it should be configurable per users choice.
I would very much appreciate that.
Is that hard to make technically? Asking as an ordinary user, I am not a programmer, I'm not able to understand the code. I'm just looking for a browser that is secure, but most comfortable for me. Of course you could say: drop Zen, stay with Waterfox. Nope. That would not be professional answer. And I need some inquiries and test in what I'm interested in.

![zen-vertical-tabs-waste-of-space](https://github.com/user-attachments/assets/7a358d6a-9f8d-4d78-ae62-b4b690e0065b)
![waterfox-horizontal-tabs-waste-of-space-NOT](https://github.com/user-attachments/assets/5e0f5f53-a8b8-4540-9833-fe2e49d30af6)

**Comment by anatasiajp** [comment]
Someone is working on Horizontal Tab: [Zen-Mod-Forbidden-Horizontal-Tabs](https://github.com/mmmintdesign/Zen-Mod-Forbidden-Horizontal-Tabs)

**Reply by rafnov** [reply]
Awesome.
Thx for the info.

**Comment by stripedgoat** [comment]
95% of websites lay out content in a way that doesn't benefit from additional horizontal space.

However, vertical space is something that the more you have of, the more content is visible at once. Thus, removing any top bars and toolbars will result in more visible content.

Having a sidebar does not decrease the visible content whatsoever, because the website's content wasn't touching the left and right edges of your viewport in the first place.

Just look at this screenshot below (or even yours). Having extra vertical space is far more beneficial than having extra horizontal space.
![Screen Shot 2024-10-17 at 2 05 34 PM](https://github.com/user-attachments/assets/a6b0ed06-2e0c-46d0-9c08-2ab30c396d6c)
Here's another screenshot. I've narrowed the window's width. I can see the same amount of content:
![Screen Shot 2024-10-17 at 2 08 01 PM](https://github.com/user-attachments/assets/3d5fa45f-b73a-43b7-bc9a-7c50b7ccf426)
And here is another screenshot, if I have a top bar:
![Screen Shot 2024-10-17 at 2 10 05 PM](https://github.com/user-attachments/assets/26533f7d-ae7b-4bf8-a315-98aba8a6335c)
As you can see, the only way I lost visible content is by compromising on the vertical space (by having a top bar).

Resizing the window in width didn't affect the amount of visible content whatsoever.

So yes - you DO maximize the amount of visible content with vertical tabs. But only if you give up having a thick top bar. That of course will waste vertical space and you will be seeing the same amount of content as with horizontal tabs. You can auto-hide your top bar, and that way you'll gain the most vertical space.

**Reply by ghostinushanka** [reply]
Here's my counter example. I do use multiple windows side by side (typically 3) and browser is often one of them.
What I (will)have lost using Zen compared to FF:
- Full visibility without touching anything for
  - Website address
  - Container tab profile
  - Search field and whatever I have searched for last time (yes, I do use a setting that does separate them to this day)
  - Plugins, cause you do click on them quite often (e.g. password manager, i've hidden it for the screenshot)

What I (will)have gained using Zen:
- Big bar stretching from top to botton that's not part of the website and just eats up valuable space for every zen instance, or,
- If it is populated with tabs, a privacy nightmare/side channel leak, cause I cannot simply screenshare rectangular part of the screen across multiple windows (left-to-right)

![ff-vs-zen](https://github.com/user-attachments/assets/b945f6d2-3dce-48c7-9a24-9503530f4820)

Having the ability to use all the above "lost" features is important. So I'd very much welcome a switch toggle for horizontal tabs and other items.
And yes, i know of tab splitting, this was one of the main reasons I came to try Zen as this is sorely missing from OG FF.
Check how does vivaldi team do tab tiling. In my case/example, I do need an hsplit, not vsplit.

**Reply by otisbartleh** [reply]
@stripedgoat 
> So yes - you DO maximize the amount of visible content with vertical tabs. But only if you give up having a thick top bar. That of course will waste vertical space and you will be seeing the same amount of content as with horizontal tabs. You can auto-hide your top bar, and that way you'll gain the most vertical space.

No, I do _**NOT**_ maximize visible content with vertical tabs, but I'm glad you know better than I do what kind of work I do and what benefits me the most.  (/sarcasm, and yes, I know you weren't talking to me directly)   Now that the sarcasm is over, I do truly mean no offense, and hope you get a chuckle out of it like I did writing it.

Monday.com, QuickBooks Online, and Google Sheets spreadsheets are 3 examples of web pages I am in _all_ day.  For each of these, vertical tabs directly eat into the visible screen space.  In Monday and Google Sheets spreadsheets, this causes a lot of side scrolling back and forth, and in QuickBooks, it makes the columns uncomfortably narrow to work with in my Invoices and Estimates, especially since I often do split view tabs, or have 2 windows up next to each other.

So while I respect your opinion for _your_ work flow, it is not the same as my work flow.  Vertical tabs are a literal deal breaker, which kind of sucks since I like everything else about Zen browser.  But I didn't know it was vertical tabs only, and was a bit annoyed to find that out after I spent an hour or so customizing it with transparency.  I also frequently switch tabs, so don't like a hidden tab bar, so hiding it isn't a viable option for my work flow.

I also respect the developers decision to only be vertical tab only, it just means I won't be using Zen Browser, bummer as it may be.  Basically, my whole point is that you shouldn't be assuming that vertical tabs are better for everyone, because they aren't.  

Plus, the pole results backup that more than twice the people prefer horizontal to vertical tabs, so there is that too.

**Comment by dalekirkwood** [comment]
![image](https://github.com/user-attachments/assets/6fcdde8b-bde9-4769-a453-a532cd0ae0c6)

I just find this ironic. I love that someone is building a plugin but this should be core functionality. I use Vivaldi now, I would love to have tried Zen but this is a killer.

**Reply by giaaaacomo** [reply]
dude the whole project is still in beta

**Reply by rafnov** [reply]
> dude the whole project is still in beta

Dude, it just got to beta 😆 i am following from early alpha 😁

**Comment by 12th-devs** [comment]
If you really must know, there is an about:config toggle for it... It is a work in progress though and I only know for sure that it is in twilight and need lots of work IMO

`zen.tabs.vertical`

**Reply by rafnov** [reply]
That does not work, just tried. Displays tabs as icons on the main bar.
Anyway, thx for the info. I didn't know that toggle.

**Reply by 12th-devs** [reply]
yeah maybe just a bit bugy (lol)
![image](https://github.com/user-attachments/assets/8ed5d3bc-a223-4f5d-9326-6a2257a557c2)

**Reply by kazajhodo** [reply]
I did this toggle, but its **extremely** buggy. That was after searching around and reading a bunch of forums to find the setting.

I can't even see tabs at all. Essential tabs are displayed, but cutoff on the bottom. Centering of the entire bar is off, all margins do not match.

![Screenshot 2025-03-03 at 15 16 39](https://github.com/user-attachments/assets/ebd5dbe1-b0a9-4c6a-8e73-7d6d196ea233)

Obviously if I can't see my tabs its a big deal. Sometimes they display and are the sidebar width. 

I'm a web dev and spend a ton of time in the browser, I always have the same tabs pinned for tracking time and various things. I often have frontend and backend open of sites, I always have two browser windows open.

The entire web, with rare exceptions, is vertically scrolled, meaning that a sidebar is an enormous waste of space. Vertical space is infinite on the web, side is finite- its that simple. However, with the option of placement- it doesn't even matter. This is FF default, so it shoud be super simple to wrap some functionality in option checks, and be off to the races.

You've made a massive opinionated mistake for no reason.

For a dev, this also consumes needed space for styling(breakpoints). Hiding key commands and auto-hide won't cut it, it wastes too much time when you add it all up.

**Comment by dalekirkwood** [comment]
> ![image](https://github.com/user-attachments/assets/6fcdde8b-bde9-4769-a453-a532cd0ae0c6)
> 
> I just find this ironic. I love that someone is building a plugin but this should be core functionality. I use Vivaldi now, I would love to have tried Zen but this is a killer.

I feel kind of bad for this comment now. I came back to Zen Browser and it's a really solid option. The fact the sidebar auto hides takes some getting used to but it offers alot more screen real estate.

**Reply by CodeCooler** [reply]
It's still a valid complaint, though. Honestly, the lack of horizontal tabs are always the first complaint I hear when trying to introduce others to Zen. It's a well-understood UX that most folks understand intuitively, and I feel not having the option prevents a lot of people from even considering its use as their daily-driver browser. I know I haven't managed to convince anyone that it's worth the mental shift yet.

**Comment by thewbuk** [comment]
Any updates on this?

**Reply by jsailo** [reply]
From all that I've read around, the developers seem to be adamant and will NOT be implementing horizontal tabs. It's still in beta and if  developers have this 'take it or leave it' attitude and WON'T listen to user feedback, I'm very skeptical of how they will take user feedback in the future when it's not in Beta. Sticking with Floorp.

**Reply by rafnov** [reply]
To be honest: my attitude changed now 😁 I'd rather vertical tabs ATM so that's why in a survey I've made option choose on demand that wins. 
I'm currently testing Helium browser and it has only horizontal tabs as a window title bar, which is really annoying. So I've installed couple of Chrome extensions that make tabs in vertical panel, but unfortunately they don't remove tabs in title bar... 
Anyway... I know that may be hard to satisfy everyone. But still: horizontal vs vertical should be up to users choice. 
Oh and btw... Years ago I've used Tree Tab (whatever it's name) in Firefox. Nowadays I don't need a tree, but afair it could hide horizontal tabs in favour of vertical panel. If extension can do such thing, why it can't be built-in function??? 

Btw. It's really lame that devs of Zen can be contacted directly only via Discord...
